src/screens/SignUp/index.tsx

```diff
@@ -1,79 +1,110 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { useTranslation } from 'react-i18next';
 import { useForm } from 'react-hook-form';
+import { useMutation } from 'react-query';
+import { Link, useNavigate } from 'react-router-dom';
 
 import LogoContainer from '../../components/LogoContainer/index';
 import { IFormSignUp } from '../../utils/types';
 import Input from '../../components/Input/index';
 import { validations } from '../../constants/formsValidation';
+import { signUp } from '../../services/UsersService';
+import Loading from '../../components/Spinner/components/loading';
+import Notification from '../../components/Notification/index';
 
 import styles from './styles.module.scss';
 
 function SignUp() {
+  const navigate = useNavigate();
   const { t } = useTranslation();
 
+  const [userError, setUserError] = useState(false);
+
   const {
     register,
     handleSubmit,
     formState: { errors },
-    watch
+    watch,
+    reset
   } = useForm<IFormSignUp>({ mode: 'onChange' });
 
+  const mutation = useMutation(signUp, {
+    onSuccess: data => {
+      console.log(data);
+      if (data.ok) {
+        reset();
+        navigate('/', { replace: true });
+        setUserError(false);
+      } else {
+        setUserError(true);
+      }
+    }
+  });
+
   const onSubmit = (data: IFormSignUp) => {
-    console.log(data);
+    setUserError(false);
+    mutation.reset();
```

Comment:
Reset should be called after the request finished, in case something needs to be fixed in the form.